extensions/integration-tests/src/utils.ts

```diff
@@ -89,3 +96,49 @@ export async function assertThrowsAsync(fn: () => Promise<any>, msg: string): Pr
 		assert.throws(f, msg);
 	}
 }
+
+/**
+ *
+ * @param databaseName name of database to check for
+ * @param ownerUri owner uri
+ * @param retryCount number of times to retry with a 5 second wait between each try
+ */
+export async function assertDatabaseCreationResult(databaseName: string, ownerUri: string, retryCount: number): Promise<void> {
+	let result: azdata.SimpleExecuteResult;
+	while (retryCount > 0) {
+		--retryCount;
+		await sleep(5000);
+
+		let query = `BEGIN TRY
+				SELECT name FROM master.dbo.sysdatabases WHERE name='${databaseName}'
+			END TRY
+			BEGIN CATCH
+				SELECT ERROR_MESSAGE() AS ErrorMessage;
+			END CATCH`;
+		result = await runQuery(query, ownerUri);
+		if (result.rowCount > 0) {
+			break;
+		}
+	}
+
+	assert(result.rowCount === 1, `Database ${databaseName} should be created`);
+	assert(result.columnInfo[0].columnName !== 'ErrorMessage', 'Checking for db creation threw error');
+}
+
+/**
+ *
+ * @param filepath File path to check for
+ * @param retryCount number of times to retry with a 5 second wait between each try
+ */
+export async function assertFileGenerationResult(filepath: string, retryCount: number): Promise<void> {
+	let exists = false;
```

Comment:
Would suggest adding a description of why these are necessary.

Reply:
Also for this one I'd suggest saying that it deletes the file too

---
In reply to: [294566924](https://github.com/microsoft/azuredatastudio/pull/6049#discussion_r294566924) [](ancestors = 294566924)